mirascope/tools/search/base.py

```diff
@@ -0,0 +1,29 @@
+from abc import ABC
+from datetime import datetime
+from typing import Generic, TypeVar
+
+from mirascope.tools.base import ConfigurableTool, _ToolConfig
+
+_ConfigT = TypeVar("_ConfigT", bound=_ToolConfig)
+_ToolSchemaT = TypeVar("_ToolSchemaT")
+
+
+class SearchToolBase(
+    ConfigurableTool[_ConfigT, _ToolSchemaT], Generic[_ConfigT, _ToolSchemaT], ABC
+):
+    """
+    Abstract base class for search-related tools.
+    Includes search history management and date-aware prompts.
+    """
+
+    @classmethod
```

Comment:
given my previous comment, this base search tool is not needed.

each tool should get it's own usage instructions (at least for now)

we can always refactor later if we find we're duplicating a lot of the same instructions